static/app/views/settings/organizationTeams/teamMembers.tsx

```diff
@@ -192,6 +168,55 @@ class TeamMembers extends Component<Props, State> {
     );
   };
 
+  removeTeamMember = (member: Member) => {
+    const {params} = this.props;
+    leaveTeam(
+      this.props.api,
+      {
+        orgId: params.orgId,
+        teamId: params.teamId,
+        memberId: member.id,
+      },
+      {
+        success: () => {
+          this.setState({
+            teamMemberList: this.state.teamMemberList.filter(m => m.id !== member.id),
+          });
+          addSuccessMessage(t('Successfully removed member from team.'));
+        },
+        error: () =>
+          addErrorMessage(
+            t('There was an error while trying to remove a member from the team.')
+          ),
+      }
+    );
+  };
+
+  updateTeamMemberRole = (member: Member, newRole: string) => {
+    const {orgId, teamId} = this.props.params;
+    const endpoint = `/organizations/${orgId}/members/${member.id}/teams/${teamId}/`;
+
+    this.props.api.request(endpoint, {
+      method: 'PUT',
+      data: {teamRole: newRole},
+      success: data => {
+        const teamMemberList: any = [...this.state.teamMemberList];
+        const i = teamMemberList.findIndex(m => m.id === member.id);
+        teamMemberList[i] = {
+          ...member,
+          teamRole: data.teamRole,
+        };
+        this.setState({teamMemberList});
```

Comment:
nit: personally i prefer if we were to pass in a refetch from the parent and re-fetch/re-pass down the list, rather than modifying the state from the response 🤷 
Ensures the parents state is properly updated as well and ensures we're sync'd with the BE